flipperdevices/flipperzero-firmware
# [FL-3677, FL-3798] RFID Improvements

# What's new

- Update RFID application GUI according to new mock-ups
- Fix crash when reading raw RFID data
- FL-3798: Fix special formatting in `text_box` widget element
- Fix crash when caused by a trailing newline in a `text_box` widget element

# Verification 

1. Compare the graphics, strings and UI flow with the mock-ups (see Jira task description). When doing so, pay extra attention for any weird formatting (rogue bold/inversed fonts, unexpected line breaks, missing text etc).
2. Make sure `Settings -> System -> Debug` is set to `ON`.
3. Go to `RFID -> Extra Actions -> Read RAW RFID Data`. The functionality should work as expected.
4. Go to `RFID -> Read` and read a card of type `Paradox` with a bad CRC.
5. The bottom line should say `CRC Mismatch, Invalid Card!`. The `!` symbol at the end should be displayed without any visual glitches.

# Checklist (For Reviewer)
